DeepLinkActivity didn't call finish() causing the app to crash

I'm trying the sample app with snapshot v2.0.0 of the library. Whenever I run the sample app through deeplink, it always crash. I think it's because the generated `DeepLinkActivity` class didn't call `finish()`. Here's the stack trace:

```
05-22 10:38:23.298 5482-5482/com.airbnb.deeplinkdispatch.sample W/Activity: An activity without a UI must call finish() before onResume() completes
05-22 10:38:23.299 5482-5482/com.airbnb.deeplinkdispatch.sample D/AndroidRuntime: Shutting down VM
05-22 10:38:23.300 5482-5482/com.airbnb.deeplinkdispatch.sample E/AndroidRuntime: FATAL EXCEPTION: main
    Process: com.airbnb.deeplinkdispatch.sample, PID: 5482
    java.lang.RuntimeException: Unable to resume activity {com.airbnb.deeplinkdispatch.sample/com.airbnb.deeplinkdispatch.DeepLinkActivity}: java.lang.IllegalStateException: Activity {com.airbnb.deeplinkdispatch.sample/com.airbnb.deeplinkdispatch.DeepLinkActivity} did not call finish() prior to onResume() completing
        at android.app.ActivityThread.performResumeActivity(ActivityThread.java:3103)
        at android.app.ActivityThread.handleResumeActivity(ActivityThread.java:3134)
        at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2481)
        at android.app.ActivityThread.-wrap11(ActivityThread.java)
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1344)
        at android.os.Handler.dispatchMessage(Handler.java:102)
        at android.os.Looper.loop(Looper.java:148)
        at android.app.ActivityThread.main(ActivityThread.java:5422)
        at java.lang.reflect.Method.invoke(Native Method)
        at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:726)
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:616)
     Caused by: java.lang.IllegalStateException: Activity {com.airbnb.deeplinkdispatch.sample/com.airbnb.deeplinkdispatch.DeepLinkActivity} did not call finish() prior to onResume() completing
        at android.app.Activity.performResume(Activity.java:6339)
        at android.app.ActivityThread.performResumeActivity(ActivityThread.java:3092)
        at android.app.ActivityThread.handleResumeActivity(ActivityThread.java:3134) 
        at android.app.ActivityThread.handleLaunchActivity(ActivityThread.java:2481) 
        at android.app.ActivityThread.-wrap11(ActivityThread.java) 
        at android.app.ActivityThread$H.handleMessage(ActivityThread.java:1344) 
        at android.os.Handler.dispatchMessage(Handler.java:102) 
        at android.os.Looper.loop(Looper.java:148) 
        at android.app.ActivityThread.main(ActivityThread.java:5422) 
        at java.lang.reflect.Method.invoke(Native Method) 
        at com.android.internal.os.ZygoteInit$MethodAndArgsCaller.run(ZygoteInit.java:726) 
        at com.android.internal.os.ZygoteInit.main(ZygoteInit.java:616) 
```

Adding `finish()` statement in its `onCreateMethod` did the work. 

Also, when will the new version be available in the JCenter repository? I can't wait to use it in my production app. 😄

--- a/deeplinkdispatch-processor/src/main/java/com/airbnb/deeplinkdispatch/DeepLinkProcessor.java
+++ b/deeplinkdispatch-processor/src/main/java/com/airbnb/deeplinkdispatch/DeepLinkProcessor.java
@@ -271,7 +271,6 @@ public class DeepLinkProcessor extends AbstractProcessor {
   }
 
   private void generateDeepLinkDelegate() throws IOException {
-
     MethodSpec notifyListenerMethod = MethodSpec.methodBuilder("notifyListener")
         .addModifiers(Modifier.PRIVATE, Modifier.STATIC)
         .returns(void.class)
@@ -420,6 +419,7 @@ public class DeepLinkProcessor extends AbstractProcessor {
         .addStatement("super.onCreate(savedInstanceState)")
         .addStatement("$T.dispatchFrom(this)",
             ClassName.get("com.airbnb.deeplinkdispatch", "DeepLinkDelegate"))
+        .addStatement("finish()")
         .build();
 
     TypeSpec deepLinkActivity = TypeSpec.classBuilder("DeepLinkActivity")

--- a/deeplinkdispatch-processor/src/test/resources/DeepLinkActivity.java
+++ b/deeplinkdispatch-processor/src/test/resources/DeepLinkActivity.java
@@ -9,5 +9,6 @@ public class DeepLinkActivity extends Activity {
   protected void onCreate(Bundle savedInstanceState) {
     super.onCreate(savedInstanceState);
     DeepLinkDelegate.dispatchFrom(this);
+    finish();
   }
 }

